quint/src/parsing/parseErrors.ts

```diff
@@ -1,19 +1,21 @@
 import { QuintError } from '../quintError'
 
-export function lowercaseTypeError(id: bigint, name: string): QuintError {
+export function lowercaseTypeError(id: bigint, name: string, prefix: string[]): QuintError {
+  const original = [...prefix, name].join('::')
+  const newName = name[0].toUpperCase() + name.slice(1)
+  const replacement = [...prefix, newName].join('::')
+
   return {
     code: 'QNT007',
     message: 'type names must start with an uppercase letter',
     reference: id,
-    data: name[0].match('[a-z]')
-      ? {
-          fix: {
-            kind: 'replace',
-            original: `type ${name[0]}`,
-            replacement: `type ${name[0].toUpperCase()}`,
-          },
-        }
-      : {},
+    data: {
+      fix: {
+        kind: 'replace',
+        original: `type ${original}`,
+        replacement: `type ${replacement}`,
+      },
```

Comment:
This information is used by the LSP as a code action to fix the problem. I don't remember in details how it works, so it's probably a good idea to test it manually.

In order to install a version of the LSP that uses your local version of Quint, you'd need `yalc`. There are some instructions here: https://github.com/informalsystems/quint/blob/main/CONTRIBUTING.md#using-the-vscode-plugin-with-an-unpublished-version-of-quint

This is for VSCode. For vim/emacs, which use the `quint-language-server` package, what I do is:
1. uninstall the package from NPM, if installed `npm uninstall @informalsystems/quint-language-server -g`
2. cd into `vscode/quint-vscode/server` and run `npm install && npm run compile && npm link`

Then, if your vim/emacs is set up to use Quint's language server, it should pick up the new version which uses your local version of Quint. And you can test writing a `type foo::bar` and checking if it gets fixed into `type foo::Bar` after calling the code action.

If you can't or don't want to do all of that, lmk. I already have most of this in place, so I can test it in a few minutes for you.

Reply:
Works as expected :)

<img width="727" alt="Screenshot 2024-09-03 at 15 27 37" src="https://github.com/user-attachments/assets/442203b8-502d-4d9f-bdd4-6a7fbaaef6b8">

Same with no prefix:

<img width="714" alt="Screenshot 2024-09-03 at 15 27 58" src="https://github.com/user-attachments/assets/0821c717-5eb3-4420-a4fd-48b055170a53">

Reply:
Only tested in Neovim so far. If you already have VS Code setup might be easier for you to give it a try. Otherwise I can always set up myself.

Reply:
yey awesome!

Reply:
No need to test it in more than one editor. All of them use the same protocol, so we are safe.